common/beerocks/bwl/CMakeLists.txt

```diff
@@ -27,21 +29,28 @@ elseif(TARGET_PLATFORM STREQUAL "openwrt")
         set(BWL_TYPE "NL80211")
     endif()
 
-    # hostapd directory
-    file(GLOB HOSTAPD_SEARCH_PATHS "${PLATFORM_BUILD_DIR}/hostapd*/hostapd-*")
+endif()
+
+if (NOT BWL_TYPE STREQUAL "DUMMY")
```

Comment:
I don't like this - it just happens that the nl80211 and dwpal types both need this, but if another bwl type is added, it will most likely not need the wpa_ctrl stuff...

I think a nicer way to organise this would be to add a new variable NEED_HOSTAPD_DIR (defaults to OFF) and set it in the different BWL_TYPE branches, and then add move this code and the rest of the code that depends on it (adding to sources and includes) after the condition.

And while you're at it: there should be an error if HOSTAPD_INCLUDE_DIR is not set.

Even better would be to use the package approach. But that's more difficult to get right.